terracotta/drivers/raster_base.py

```diff
@@ -36,6 +38,42 @@
 logger = logging.getLogger(__name__)
 
 
+class LFUCacheWithCompression(LFUCache):
+    """ Least-frequently-used Cache with data compression
+    """
+
+    def __getitem__(self, key: Any) -> np.ma.MaskedArray:
+        compressed_item = super().__getitem__(key)
+        return self._decompress_masked_array(compressed_item)
+
+    def __setitem__(self, key: Any, value: np.ma.MaskedArray) -> None:
+        super().__setitem__(key, self._compress_masked_array(value))
+
+    def _compress_masked_array(self, array: np.ma.MaskedArray) -> Tuple:
+        compressed_data = zlib.compress(array.data)
+        mask_to_int = np.packbits(array.mask.astype(np.uint8))
+        compressed_mask = zlib.compress(mask_to_int)
+        return (compressed_data,
+                compressed_mask,
+                array.dtype.name,
+                array.shape
+                )
+
+    def _decompress_masked_array(self, data: Tuple) -> np.ma.MaskedArray:
+        dt = np.dtype(data[2])
+        shape = data[3]
+        decompressed_data = np.frombuffer(zlib.decompress(data[0]), dtype=dt).reshape(shape)
+        decompressed_mask_packed = np.frombuffer(zlib.decompress(data[1]), dtype=np.uint8)
+        decompressed_mask_unpacked = np.unpackbits(decompressed_mask_packed)[:np.prod(shape)]
+        decompressed_mask = decompressed_mask_unpacked.reshape(shape)
+        return np.ma.masked_array(decompressed_data, mask=decompressed_mask)
+
+
+def _get_size_of(x: Tuple) -> int:
+    sizes = map(sys.getsizeof, x)
```

Comment:
I have found `sys.getsizeof` to be blatantly wrong for NumPy arrays (which is why I used `nbytes` instead). Just a heads-up.

Reply:
Interesting. I guess we could change this to just add the `nbytes` together, the size of the dtype str and shape should be insignificant.

Reply:
That sounds more foolproof to me.

Reply:
Would even deserve a unit test to prevent this from breaking in the future. We have had problems with the cache not working as intended because of this.

Reply:
I used `sys.getsizeof` because we want to find the size of two compressed objects and not numpy arrays. I tested and seems to works fine on compressed objects, whereas in arrays or dictionaries it doesn't.

Reply:
Since you already checked it, could you please add a unit test? :) Just a simple sanity check, doesn't need to be accurate to the byte.

Reply:
Yes, seems we forgot that the compressed array are not numpy arrays 😄
They are probably some kind of Bytes object, right? So if `sys.getsizeof` should not work for whatever reason, a simple `len()` might do it.